core/Mail.php

```diff
@@ -52,6 +53,23 @@ public function setDefaultFromPiwik()
         $this->setFrom($fromEmailAddress, $fromEmailName);
     }
 
+    public function setWrappedHtmlBody(View $body)
+    {
```

Comment:
Just wondering... should this be maybe in a new class which can be injected etc? Then we could also replace it through DI etc... I'd say the class does otherwise maybe too many things like formatting plus setting mail etc.

Reply:
It's here so derived classes can use it.

The idea in this PR is that future emails will create their own class that derived from Mail, then in event handlers plugins can detect exactly which email is about to be sent. The derived class would call this method to set the HTML body. So the methods in this class should done one of two things: configure the email to be sent, or send the email.

Reply:
I was more thinking if it was a class, we could overwrite the wrap globally for all emails. For example on Cloud.

Reply:
I see, overwrite just the wrapping logic, gotcha.